group/views.py

```diff
@@ -20,3 +23,89 @@ def create_group(request):
             return Response(serializer.data, status=status.HTTP_201_CREATED)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
     return None
+
+
+@api_view(['POST'])
+@permission_classes([IsAuthenticated])
+def add_users(request, group_id):
+    """Add user(s) to a group"""
+    if request.method == 'POST':
+        data = request.data.copy()
+
+        data.pop('creator', None)
+        data.pop('name', None)
+        try:
+            group = Group.objects.get(id=group_id)
+        except Group.DoesNotExist:
+            return Response({"error": "Group not found"},
+                            status=status.HTTP_404_NOT_FOUND)
+        errors = []
+        for member in group.members.all():
```

Comment:
Hmm how are you accessing group outside the try block? Also the logic in this for loop is quite confusing, what are you trying to achieve?

Reply:
This loop ensures that members not in the current group are appended to a list. This step prevents overriding already existing members and ensures that only new members are added.